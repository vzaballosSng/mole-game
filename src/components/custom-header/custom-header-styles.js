import { css } from 'lit';

export const customHeaderStyles = css`
	:host {
    display: block;
    box-sizing: border-box;
    background-color: #a0522d;
    padding: 1.25rem;
    width: 100%;
	}

  .header {
    display: flex;
    justify-content: space-between;
    padding: 0;
  }
`;
