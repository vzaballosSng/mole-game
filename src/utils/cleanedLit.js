// eslint-disable-next-line no-undef
const { prettyDOM } = require('@testing-library/dom');

export const cleanedLit = (htmlElement) => {
	const domHTML = prettyDOM(htmlElement, Infinity);
	const cleaned = domHTML.replace(/"<!--\?lit\$.*?-->"/g, '""');
	return cleaned;
};
export const cleanedLitElements = (htmlElements) => {
	const cleaned = htmlElements.map((htmlElement) => {
		return cleanedLit(htmlElement);
	});
	return cleaned;
};
