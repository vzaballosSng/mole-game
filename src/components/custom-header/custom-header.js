import { LitElement, html } from 'lit';
import { customHeaderStyles } from './custom-header-styles.js';
import '../custom-select/custom-select.js';

export class CustomHeader extends LitElement {
	static get properties() {
		return {
			playerName: { type: String, attribute: 'player-name' },
		};
	}

	static get styles() {
		return [customHeaderStyles];
	}

	constructor() {
		super();
    this.playerName = '';
	}

	render() {
		return html`
		<header class="header">
      <slot name="left-side"></slot>
      <slot name="right-side"></slot>
		</header>
		`;
	}
}

window.customElements.define('custom-header', CustomHeader);
