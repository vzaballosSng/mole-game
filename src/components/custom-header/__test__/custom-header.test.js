import { html, fixture } from '@open-wc/testing';
import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../utils/cleanedLit.js';
import '../custom-header.js';

expect.extend(toHaveNoViolations);

describe('custom-header', () => {
	it('matches the snapshot', async () => {
		const el = await fixture(
			html`<custom-header></custom-header>
			>`
		);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});

	it('accessibility test', async () => {
		const el = await fixture(html`<custom-header></custom-header>`);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
