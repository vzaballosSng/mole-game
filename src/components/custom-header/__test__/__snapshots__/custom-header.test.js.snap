// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-header matches the snapshot 1`] = `"[36m<custom-header />[39m"`;
